### ISSUE TITLE
libQGLViewer headers are only partially copied by cmake --install

### ISSUE THREAD

**Issue by dennis2society** [sofa-framework/sofa#4196]
### Problem

**Description**
After building Sofa from source and running ```cmake --install build/``` there are headers missing for QGLViewer.
The installation also adds some redundant QGLViewer directories to the install path.

**Steps to reproduce**
- clone the current Sofa repo
- cd into the cloned folder
- run ```cmake -B build/``` 
- run ```cmake --build build/```
- run ```cmake --install build/```

The last line installs everything into $sofa_dir/build/install/.
For QGLViewer this results in 
1. an overly long install path:
    - ```$sofa_dir/build/install/include/extlibs/QGLViewer/QGLViewer/QGLViewer```
2. missing headers from libQGLViewer, only the following ones are available, most notably qglviewer.h is missing here:
   - ```config.h  constraint.h  domUtils.h  mouseGrabber.h  quaternion.h  ui_ImageInterface.h  ui_VRenderInterface.h  vec.h```
    
**Expected behavior**
- All headers for libQGLViewer should be copied to the install directory
- The path for the headers should be something like ```$sofa_dir/build/install/include/extlibs/QGLViewer/include/QGLViewer/*.h``` or ```$sofa_dir/build/install/include/extlibs/QGLViewer/*.h```

---------------------------------------------

### Environment

**Context**
This happens on both Linux and Windows but only became critical on Windows when I had tried to use the Sofa provided libQGLViewer in my own project. This could temporarily be fixed by changing my own CMake include path to ```{SOFA_INSTALL_DIR}/../../Sofa/GUI/Qt/libQGLViewer```

**Comment by dennis2society** [comment]
I have identified the actual issue in ```$sofa_dir/Sofa/GUI/Qt/libQGLViewer/QGLViewer/CMakeLists.txt```:
- line 124 has a few mentions of ```/${PROJECT_NAME}``` too much, removing these at least reduces the header install path to ```$sofa_dir/build/install/include/extlibs/QGLViewer/QGLViewer/*.h```
- for some reason the ```MOC_HEADER_FILES``` (line 21 ff.) are not considered for copying by your ```sofa_create_package_with_targets``` macro. For testing I have simply copied the headers listed in ```MOC_HEADER_FILES``` to the "normal" ```HEADER_FILES```.  This does not affect the build and finally copies the previously missing MOC headers to the install include directory.

I can make a pull request for this. Or I can wait a bit if someone has an idea on how to properly copy/install the MOC_HEADERS without the redundant mentioning in both header lists.
